lib/faraday/deprecate.rb

```diff
@@ -12,9 +12,13 @@ def self.proxy_class(new_klass)
         class << k
           extend Faraday::Deprecate
           # Make this more human readable than #<Class:Faraday::ClientError>
-          klass_name = superclass.to_s[/^#<Class:(\w*::\w*)>$/, 1]
+          klass_name = superclass.to_s[/^#<Class:(\w{1}[\w:]*)>$/, 1]
```

Comment:
```suggestion
          klass_name = superclass.to_s[/^#<Class:([\w:]+)>$/, 1]
```

Can you think of a case this wouldn't work for?

Reply:
Yeah, that looks fine. I was thinking the name should start with a letter, but that's silly considering we have no control over what ruby decides the superclass name will be.